### PR TITLE
Add model input script and provide an example of the reusable model action

### PR DESCRIPTION
**Pull request by venexia**
- Update `active_analyses.rds` so that it is compatible with reusable action
- Add `make_model_input.R` to prepare input files for reusable action
- Update YAML to include actions for `make_model_input.R`
- Update YAML to include an example of the reusable action

### CODE REVIEW

**Review by josephignace — APPROVED** [review]

File: analysis/model/make_model_input.R [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+  
+  # Remove outcomes outside of follow-up time ------------------------------------
+  print('Remove outcomes outside of follow-up time')
+  
+  input <- dplyr::rename(input, 
+                         "out_date" = active_analyses$outcome[i],
+                         "exp_date" = active_analyses$exposure[i])
+  
+  input <- input %>% 
+    dplyr::mutate(out_date = replace(out_date, which(out_date>end_date | end_date<index_date), NA),
```
[the remark this answers is not in the document]
Comment (venexia) [reply analysis/model/make_model_input.R]: Yes, thanks! Fixed here: https://github.com/opensafely/post-covid-mentalhealth/pull/20/commits/b3dbdf0f83cf3efb544a59ed5319ac2e4a55071e

File: analysis/model/make_model_input.R [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+  }
+  
+  # Make model input: sub_covid_hospitalised -------------------------------------
+  
+  if (active_analyses$analysis[i]=="sub_covid_hospitalised") {
+    
+    print('Make model input: sub_covid_hospitalised')
+    
+    df <- input %>% 
+      dplyr::mutate(end_date = replace(end_date, which(sub_cat_covid19_hospital=="non_hospitalised"), exp_date),
```
[the remark this answers is not in the document]
Comment (venexia) [reply analysis/model/make_model_input.R]: Thanks again. Changes here: https://github.com/opensafely/post-covid-mentalhealth/pull/20/commits/4bfa54c232c90beb3cfdae7f4aeff37bd0dc8c9f

File: analysis/model/make_model_input.R [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+    write.csv(x = df, file = paste0("output/model_input-",active_analyses$name[i],".csv"), row.names=FALSE)
+    print(paste0("Saved: output/model_input-",active_analyses$name[i],".csv"))
+    rm(df)
+    
+  }
+  
+  # Make model input: sub_covid_hospitalised -------------------------------------
+  
+  if (active_analyses$analysis[i]=="sub_covid_hospitalised") {
+
```
[the remark this answers is not in the document]
Comment (venexia) [reply analysis/model/make_model_input.R:120]: https://github.com/opensafely/post-covid-mentalhealth/pull/20/commits/71ca6390f3ce179d8ed09775cdd15d83542efcec

File: analysis/model/make_model_input.R [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+  }
+  
+  # Make model input: sub_sex_female ---------------------------------------------
+  
+  
+  if (active_analyses$analysis[i]=="sub_sex_female") {
+    
+    print('Make model input: sub_sex_female')
+    
+    df <- input[input$sub_bin_covid19_confirmed_history==TRUE &
```
[the remark this answers is not in the document]
Comment (venexia) [reply analysis/model/make_model_input.R]: https://github.com/opensafely/post-covid-mentalhealth/pull/20/commits/71ca6390f3ce179d8ed09775cdd15d83542efcec

**Pull request merged**